tools/input-segments/src/InputSegments.ts

```diff
@@ -415,163 +508,416 @@ export class InputSegments extends TextfieldBase {
-                hour: '2-digit',
-                ...(includeMinutes && { minute: '2-digit' }),
-                ...(includeSeconds && { second: '2-digit' }),
-            };
+            return;
         }
 
-        this.formatter = new DateFormatter(this.locale, {
-            ...dateOptions,
-            ...timeOptions,
-        });
+        if (time !== undefined) {
+            this.newDateTime = this.dateToCalendarDateTime(time);
+        }
+
+        // If date segments are being used, we need to change the date part to use the value of these segments
+        if (this.includeDate && date !== undefined) {
+            const dateCalendar = this.dateToCalendarDateTime(date);
+
+            this.newDateTime = this.newDateTime?.set({
+                year: dateCalendar.year,
+                month: dateCalendar.month,
+                day: dateCalendar.day,
+            });
+        }
     }
 
     /**
-     * * Defines the number parser using the defined locale
+     * Emits the new value for date/time if it is already defined
      */
-    private setNumberParser(): void {
-        this.numberParser = new NumberParser(this.locale, {
-            maximumFractionDigits: 0,
-        });
+    protected emitNewDateTime(): void {
+        const dateTime = this.newDateTime
+            ? this.newDateTime.toDate(this.timeZone)
+            : undefined;
+
+        this.dispatchEvent(
+            new CustomEvent('change', {
+                bubbles: true,
+                composed: true,
+                detail: dateTime,
+            })
+        );
     }
 
     /**
-     * If a datetime is received by the component via property, it will use it as the current datetime to render the
-     * input
+     * The parts returned by the `formatToParts()` function of `Intl.DateTimeFormat` have only two properties, `type`
+     * and `value`, but we need more information for each segment, so we convert it to the type we need
+     *
+     * @param part - Part/segment to be “translated” (mapped)
      */
-    private setCurrentDateTime(): void {
-        if (!this.selectedDateTime) {
-            return;
+    protected mapToSegment(part: Intl.DateTimeFormatPart): Segment {
+        const type = part.type;
+        const formatted = part.value;
+
+        if (type === 'literal') {
+            return {
+                type,
+                formatted,
+            };
         }
 
-        this.selectedDateTime = new Date(this.selectedDateTime);
+        const { value, minValue, maxValue } = this.getValueAndLimits(type);
+        const placeholder = this.getPlaceholder(type, part.value);
 
-        if (!this.isValidTime(this.selectedDateTime)) {
-            this.selectedDateTime = undefined;
-            return;
+        const segment: Segment = {
+            type,
+            formatted,
+            ...(value !== undefined && { value }),
+            ...(minValue !== undefined && { minValue }),
+            ...(maxValue !== undefined && { maxValue }),
```

Comment:
Is it possible to get the same here via:
```suggestion
            ...this.getValueAndLimits(type)
```
As `undefined` destructures nothing into the object?